### PR TITLE
refactor: edv client and add HeadersOpt optional function

### PR DESCRIPTION
**Pull request by fqutishat**
Signed-off-by: Firas Qutishat <firas.qutishat@securekey.com>

### CODE REVIEW

**Review by rolsonquadras — APPROVED** [review]

File: pkg/client/client.go [hunk 1]
```diff
@@ -49,18 +50,25 @@ func WithTLSConfig(tlsConfig *tls.Config) Option {
 	}
 }
 
-// Opts is used to interact with an EDV operation.
-type Opts struct {
-	httpHeader map[string]string
+// WithHeaders option is for setting additional http request headers (since it's a function, it can call a remote
```
Comment (DRK3) [inline pkg/client/client.go:53]: Unclosed bracket in comment

File: pkg/client/client.go [hunk 2]
```diff
@@ -49,18 +50,25 @@ func WithTLSConfig(tlsConfig *tls.Config) Option {
...
+}
 
-// WithHTTPHeader option is
-func WithHTTPHeader(key, value string) EDVOption {
-	return func(opts *Opts) {
-		opts.httpHeader[key] = value
+// ReqOption edv req option
+type ReqOption func(opts *ReqOpts)
+
+// WithRequestHeader option is for setting additional http request headers (since it's a function, it can call a remote
```
Comment (DRK3) [inline pkg/client/client.go:68]: Unclosed bracket here too. Is the comment cut off?

Comment (fqutishat) [reply pkg/client/client.go]: done

File: pkg/client/client.go (hunk 1 above)
Comment (fqutishat) [reply pkg/client/client.go]: done

**Review by sudeshrshetty — APPROVED** [review]

File: pkg/client/client.go [hunk 3]
```diff
@@ -230,128 +239,111 @@ Response body: %s`, endpoint, jsonToSend, resp.StatusCode, respBytes)
...
 	respBytes, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
-		return fmt.Errorf("failed to read response body: %w", err)
+		return -1, nil, nil, err
 	}
 
-	logger.Debugf(`Sent DELETE request to %s.
-Response status code: %d
-Response body: %s`, fullEndpoint, resp.StatusCode, respBytes)
+	logger.Debugf(`sent GET request to %s response status code: %d response body: %s`, endpoint,
```
Comment (DRK3) [inline pkg/client/client.go:333]: These aren't always GET requests - you need to include the method type here

Comment (fqutishat) [reply pkg/client/client.go]: @DRK3 done

File: pkg/client/client.go [hunk 4]
```diff
@@ -230,128 +239,111 @@ Response body: %s`, endpoint, jsonToSend, resp.StatusCode, respBytes)
...
 	respBytes, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
-		return fmt.Errorf("failed to read response body: %w", err)
+		return -1, nil, nil, err
 	}
 
-	logger.Debugf(`Sent DELETE request to %s.
-Response status code: %d
-Response body: %s`, fullEndpoint, resp.StatusCode, respBytes)
+	logger.Debugf(`sent %s request to %s response status code: %d response body: %s`, method, endpoint,
```
Comment (DRK3) [inline pkg/client/client.go:333]: Any reason for the removal of the new lines in the debug message? I'm afraid that it's going to be very difficult to read this log output.

**Review by DRK3 — APPROVED** [review]
Left one more note regarding the removal of line breaks from log output, but it's not a major issue

File: pkg/client/client.go (hunk 3 above)
Comment (Baha-sk) [reply pkg/client/client.go]: `Get` can easily be replaced with %s and `req.Method`

**Review by Baha-sk — APPROVED** [review]

File: pkg/client/client.go (hunk 4 above)
Comment (Baha-sk) [reply pkg/client/client.go:333]: @DRK3 ops don't like multi line logs, it makes it difficult searching/troubleshooting entries

Comment (fqutishat) [reply pkg/client/client.go:333]: feedback from dev ops: new lines in debug is nightmare

Comment (DRK3) [reply pkg/client/client.go:333]: OK! I'll avoid them from now on.

**Comment by Baha-sk** [comment]
> Left one more note regarding the removal of line breaks from log output, but it's not a major issue

as a rule of thumb, logs should not have line breaks @DRK3 to help ease the life of ops folks ;-)

**Pull request merged**